File: api/app/Module/Db/Dao/Auth/Scene.php
<?php

declare(strict_types=1);

namespace App\Module\Db\Dao\Auth;

use App\Module\Db\Dao\AbstractDao;

/**
 * @property int $sceneId 权限场景ID
 * @property string $sceneCode 标识（代码中用于识别调用接口的所在场景，做对应的身份鉴定及权力鉴定。如已在代码中使用，不建议更改）
 * @property string $sceneName 名称
 * @property string $sceneConfig 配置（内容自定义。json格式：{"alg": "算法","key": "密钥","expTime": "签名有效时间",...}）
 * @property int $isStop 是否停用：0否 1是
 * @property string $updateTime 更新时间
 * @property string $createTime 创建时间
 */
class Scene extends AbstractDao
{
    /**
     * 解析insert（独有的）
     *
     * @param string $key
     * @param [type] $value
     * @param integer $index
     * @return boolean
     */
    protected function insertOfAlone(string $key, $value, int $index = 0): bool
    {
        switch ($key) {
            case 'sceneConfig':
                if (!($value === '' || $value === null)) {
                    $this->insert[$index][$key] =  is_array($value) ? json_encode($value, JSON_UNESCAPED_UNICODE) : $value;
                }
                return true;
        }
        return false;
    }

    /**
     * 解析update（独有的）
     *
     * @param string $key
     * @param [type] $value
     * @return boolean
     */
    protected function updateOfAlone(string $key, $value = null): bool
    {
        switch ($key) {
            case 'sceneConfig':
                if ($value === '' || $value === null) {
                    $this->update[$this->getTable() . '.' . $key] = null;
                } else {
                    $this->update[$this->getTable() . '.' . $key] = is_array($value) ? json_encode($value, JSON_UNESCAPED_UNICODE) : $value;
                }
                return true;
        }
        return false;
    }

    /**
     * 解析where（独有的）
     *
     * @param string $key
     * @param string|null $operator
     * @param [type] $value
     * @param string|null $boolean
     * @return boolean
     */
    protected function whereOfAlone(string $key, string $operator = null, $value, string $boolean = null): bool
    {
        switch ($key) {
            case 'sceneName':
                if ($operator === null) {
                    $this->where[] = ['method' => 'where', 'param' => [$this->getTable() . '.' . $key, 'like', '%' . $value . '%', $boolean ?? 'and']];
                } else {
                    $this->where[] = ['method' => 'where', 'param' => [$this->getTable() . '.' . $key, $operator, $value, $boolean ?? 'and']];
                }
                return true;
        }
        return false;
    }
}

File: api/api/auth/internal/scene.go
package api

type ReqSceneList struct {
	Filter ReqSceneListFilter `p:"filter"  v:""`
	Field  []string           `p:"field"  v:"foreach|min-length:1"`
	Order  []string           `p:"order"  v:""`
	Page   uint               `p:"page"  v:"min:1"`
	Limit  uint               `p:"limit"  v:""` //传0取全部
}

type ReqSceneListFilter struct {
	SceneId     string // 权限场景ID
	SceneCode   string // 标识（代码中用于识别调用接口的所在场景，做对应的身份鉴定及权力鉴定。如已在代码中使用，不建议更改）
	SceneName   string // 名称
	SceneConfig string // 配置（内容自定义。json格式：{"alg": "算法","key": "密钥","expTime": "签名有效时间",...}）
	IsStop      string // 是否停用：0否 1是
	UpdateTime  string // 更新时间
	CreateTime  string // 创建时间
	/* SceneId:     "sceneId",
	            SceneCode:   "sceneCode",
	            SceneName:   "sceneName",
	            SceneConfig: "sceneConfig",
	            IsStop:      "isStop",
	            UpdateTime:  "updateTime",
	            CreateTime:  "createTime",
		SceneId     *uint  `c:"sceneId,omitempty" p:"sceneId" v:"min:1"`
		SceneName   string `c:"sceneName,omitempty" p:"sceneName" v:"length:1,30|regex:^[\\p{L}\\p{M}\\p{N}_-]+$"`
		SceneCode   string `c:"sceneCode,omitempty" p:"sceneCode" v:"length:1,30|regex:^[\\p{L}\\p{M}\\p{N}_-]+$"`
		IsStop      *uint  `c:"isStop,omitempty" p:"isStop" v:"in:0,1"`
		SceneConfig string `c:"sceneConfig,omitempty" p:"sceneConfig" v:"json"` */
}
